distributed/worker_state_machine.py

```diff
@@ -88,8 +88,6 @@
     "constrained",
     "executing",
     "long-running",
-    "cancelled",
-    "resumed",
```

Comment:
This change triggers a different decision in https://github.com/dask/distributed/blob/fae59c46d5ba66527840cca0287eada1d668dc65/distributed/worker_state_machine.py#L1960-L1965

I think this slipped in and I didn't intend for this to persist for the PR. This is how I started off but the actual change I'm looking for is the removal of `is_protected`.

I'll need to look a bit more closely into this transition

Reply:
`not in READ | PROCESSING | {"memory"}`
translates to
- main: `{'error', 'fetch', 'flight', 'forgotten', 'missing', 'released', 'rescheduled'}`
- This PR `{'error', 'fetch', 'flight', 'forgotten', 'missing', 'released', 'rescheduled', 'resumed', 'cancelled'}`

Reply:
This is a side-effect, but it is getting us closer to removing the `cancelled`/`resumed` states. See the changes in the story for `test_resumed_cancelled_handle_compute`:
https://github.com/dask/distributed/pull/7490/files#diff-547f8648dc3033245c94f246de2eebede153b55ce604f9c04ad1265cd7e85f11L538-R540

https://github.com/dask/distributed/pull/7490/files#diff-547f8648dc3033245c94f246de2eebede153b55ce604f9c04ad1265cd7e85f11L561-R571

Effectively, we cancel a `resumed` task if it's a dependency of a task we want to forget. From my perspective, this is generally closer to the desired behavior. Seeing `resumed` as the equivalent of `executing`, this is now the worker-side equivalent of of releasing a `processing` task if it has been a dependency of a task we wanted to forget. Since the scheduler is not aware of the worker sneakily processing the task without telling anybody, this cancellation needs to happen on the worker.

Reply:
The unit tests looks fine.

However, I don't know for sure right now if the scheduler truly has no way to provoke this path

> Effectively, we cancel a resumed task if it's a dependency of a task we want to forget.

yes, that's fine. The problem is that resumed is a degenerate state and I was worried about a `resumed(flight->waiting)`

I'm also surprised that it is no longer possible for a `resumed(executing->fetch) => waiting` transition. That's effectively a case where the task is supposed to be fetched by a worker but the scheduler changes its mind and reschedules it on this worker instead.

Reply:
At the time of rescheduling the scheduler doesn't know that this worker is aware of that task and it typically doesn't send a release message, i.e. it should still be possible to trigger the above transition.
I'm worried that we are missing this case

Reply:
Are you worried about our handling of the test or whether we might be lacking a test? If that transition gets triggered, the task should be transitioned to `executing` and subsequently be refused by the scheduler due to an out-of-date `run_id`. I agree that we should double-check if there's a test though.

Reply:
saw the comment too late. I was worried about this transition still being reachable but us not covering it. Turns out this is unreachable which is great